PWGHF/HFL/Tasks/taskSingleMuon.cxx

```diff
@@ -171,15 +173,23 @@ struct HfTaskSingleMuonSelection {
     HistogramConfigSpec hTHnPt{HistType::kTHnSparseD, {axispT, axispTGen, axispTDif, axisTrkType}, 4};
     HistogramConfigSpec hTHnEta{HistType::kTHnSparseD, {axisEta, axisEtaGen, axisEtaDif, axisTrkType}, 4};
     HistogramConfigSpec hTHnP{HistType::kTHnSparseD, {axisP, axisPGen, axisPDif, axisTrkType}, 4};
+    HistogramConfigSpec hTHnMuLite{HistType::kTHnSparseD, {axispT, axisEta, axisDCA, axisTrkType, axisChi2MatchMCHMFT}, 5};
 
-    registry.add("hMuBcuts", "", hTHnMu);
-    registry.add("hMuAcuts", "", hTHnMu);
-    registry.add("hPTBcuts", "", hTHnPt);
-    registry.add("hPTAcuts", "", hTHnPt);
-    registry.add("hEtaBcuts", "", hTHnEta);
-    registry.add("hEtaAcuts", "", hTHnEta);
-    registry.add("hPBcuts", "", hTHnP);
-    registry.add("hPAcuts", "", hTHnP);
+    if (fillLiteHist) {
+      registry.add("hMuLiteBeforeCuts", "", hTHnMuLite);
+      registry.add("hMuLiteAfterCuts", "", hTHnMuLite);
+    } else {
```

Comment:
There seems to be a problem with the `else` condition here. You create those histograms only if `!fillLiteHist` but in `runMuonSelMC` you fill them regardless of the value of `fillLiteHist`. So if you run `runMuonSelMC` with `fillLiteHist = true`, you will attempt to fill a non-existent histogram.

Reply:
Also please rename `runMuonSelMC` to `runMuonSelMc`